### PR TITLE
Erace warnings with adding test builder

### PR DESCRIPTION
**Pull request by udzura**
I have added:
- travis.yml config
- local test build tasks
- ruby-layer smoke test

And I have:
- added `const` keyword in proper place
- set config that ignores C90-ish warning

Then I got no warnings with:
- CentOS Linux release 7.2.1511 (Core)
- gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-4)

### CODE REVIEW

**Comment by udzura** [comment]
Passed!

**Comment by udzura** [comment]
This PR seems to skip `-Wall` unintendedly.... I will fix

File: .travis-build_config.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+MRuby::Build.new do |conf|
+  toolchain :gcc
+  conf.gembox 'default'
+  conf.gem File.expand_path(File.dirname(__FILE__))
+  conf.enable_test
+
+  conf.cc do |cc|
+    cc.flags = ["-Wno-error=declaration-after-statement"]
```
Comment (matsumotory) [inline .travis-build_config.rb]: `cc.flags` already have flags array. So, you should use `<<` method.

**Comment by matsumotory** [comment]
Thank you for the patch. I'll merge this if you fix my identification https://github.com/matsumoto-r/mruby-capability/pull/3/files#r68010853.

File: .travis-build_config.rb (hunk 1 above)
Comment (udzura) [inline .travis-build_config.rb]: That's it...

**Comment by udzura** [comment]
```
/Users/udzura/.ghq/github.com/udzura/mruby-capability/src/mrb_capability.c:51:22: 警告: 使用されない変数 ‘ctx’ です [-Wunused-variable]
     mrb_cap_context *ctx = (mrb_cap_context *)p;
```

This remain, but is looking like WIP code. So leave this OK.

**Comment by udzura** [comment]
Passed  again!!

**Comment by matsumotory** [comment]
Thanks!!

**Pull request merged**